src/init.cpp

```diff
@@ -1356,15 +1355,27 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
     // is not yet setup and may end up being set up twice if we
     // need to reindex later.
 
+    // see Step 2: parameter interactions for more information about these
+    fListen = args.GetBoolArg("-listen", DEFAULT_LISTEN);
+    fDiscover = args.GetBoolArg("-discover", true);
+    g_relay_txes = !args.GetBoolArg("-blocksonly", DEFAULT_BLOCKSONLY);
+
     assert(!node.banman);
     node.banman = MakeUnique<BanMan>(GetDataDir() / "banlist.dat", &uiInterface, args.GetArg("-bantime", DEFAULT_MISBEHAVING_BANTIME));
     assert(!node.connman);
     node.connman = MakeUnique<CConnman>(GetRand(std::numeric_limits<uint64_t>::max()), GetRand(std::numeric_limits<uint64_t>::max()), args.GetBoolArg("-networkactive", true));
 
+    // Don't initialize fee estimation with old data if we don't relay transactions,
+    // as they would never get updated.
+    assert(!node.feeEstimator);
+    if (g_relay_txes) {
```

Comment:
Consider joining into one line for simple one-statement if blocks:

```suggestion
    if (g_relay_txes) node.feeEstimator = std::make_shared<CBlockPolicyEstimator>();
```